File: src/main/java/com/example/Dashboard/model/Venda.java
package com.example.Dashboard.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.time.ZoneId;
import java.time.ZonedDateTime;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Venda {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String produto;
    private String categoria;
    private int quantidade;
    private double valor;

    private String cidade;
    private String estado;
    private String pais;
    private String cep;

    private Double latitude;  
    private Double longitude;

    @Column(name = "data_venda", nullable = false, updatable = false)
    private LocalDateTime dataVenda;

@PrePersist
protected void onCreate() {
    this.dataVenda = ZonedDateTime.now(ZoneId.of("America/Sao_Paulo")).toLocalDateTime();
}
}

File: models/Venda.js
class Venda {
  constructor(data) {
    this.id = data.id;
    this.produto = data.produto;
    this.categoria = data.categoria;
    this.quantidade = data.quantidade;
    this.valor = data.valor;
    this.cidade = data.cidade;
    this.estado = data.estado;
    this.pais = data.pais;
    this.cep = data.cep;
    this.latitude = data.latitude;
    this.longitude = data.longitude;
    this.dataVenda = data.dataVenda;
  }

  static fromDatabase(row) {
    return new Venda({
      id: row.id,
      produto: row.produto,
      categoria: row.categoria,
      quantidade: row.quantidade,
      valor: row.valor,
      cidade: row.cidade,
      estado: row.estado,
      pais: row.pais,
      cep: row.cep,
      latitude: row.latitude,
      longitude: row.longitude,
      dataVenda: row.data_venda
    });
  }
}

module.exports = Venda;